### PR TITLE
Added counting-sort.md file

### LINKED ISSUE #501: [General] <Add counting sort algorithm>
### Issue Description
I would like to add a new sorting algorithm to our collection: Counting Sort. This algorithm is particularly efficient for sorting integers within a known, limited range. It operates in $O(n + k)$ time complexity, making it a great choice for specific scenarios.

### Additional Context
Add any other context, code snippets, or screenshots related to the issue or question.

---

**Would you like to work on this issue?**  
- [Yes ] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by kjl98**
### Description
Counting Sort is a non-comparative sorting algorithm used for sorting integers when the input values are within a specific range. It works by counting the occurrences of each unique element in the input array and using these counts to determine the positions of each element in the sorted output.

How It Works:
1) Count Array: Create an array (count[]) to store the count of each unique value from the input array.

2) Count Elements: Traverse the input array and increment the count for each value.

3) Cumulative Counts: Modify the count array to store the cumulative counts, which helps in placing elements in the correct positions in the sorted array.

4) Build Output: Use the count array to place each element in the sorted order in the output array.

5) Copy Back: Optionally, copy the sorted elements back to the original array.

Fixes #501 

### Type of change: Counting Sort. This algorithm is particularly efficient for sorting integers within a known, limited range. It operates in $O(n + k)$ time complexity, making it a great choice for specific scenarios.

Fixes #501 
I would like to add a new sorting algorithm to our collection: Counting Sort. This algorithm is particularly efficient for sorting integers within a known, limited range. It operates in O(n + k) time complexity, making it a great choice for specific scenarios.

Fixes #501 

### Type of change
Fixes #501 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/basic-dsa/array/counting-sort.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Counting Sort Algorithm
```
Comment (ajay-dhangar) [inline docs/basic-dsa/array/counting-sort.md]: We need on top

```md
---
id: ___
title: ___
sidebar_label: ___
sidebar_position: ___
description: ___
tags: [___, __, ]
---
```

### Detailed Guidelines

- id: Unique identifier for the route.
- title: Title of the problem.
- sidebar_label: Problem number or label for the sidebar.
- tags: Relevant tags for the problem.
- description: A short SEO-friendly description. No HTML tags.

File: docs/basic-dsa/array/counting-sort.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+## Overview
+Counting Sort is an efficient, non-comparative sorting algorithm suitable for sorting integers within a specific range. It operates by counting the occurrences of each unique value in the input data and then using those counts to determine the positions of each value in the sorted output.
+
+## Characteristics
+- **Time Complexity**: O(n + k), where `n` is the number of elements in the input array and `k` is the range of the input values.
```
Comment (ajay-dhangar) [inline docs/basic-dsa/array/counting-sort.md]: replace `O(n + k)` with `$O(n + k)$`

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update code based on suggestions

File: docs/basic-dsa/array/counting-sort.md (hunk 1 above)
Comment (kjl98) [reply docs/basic-dsa/array/counting-sort.md:10]: okay

**Comment by kjl98** [comment]
Sir, I've successfully merged these changes in my forked repo, now how can I create a new pull request to you for merging my changes?

**Comment by kjl98** [comment]
And sir please let me know if I made any mistake in these changes

File: docs/basic-dsa/array/counting-sort.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
+---
+id: @kjl98
```
Comment (ajay-dhangar) [inline docs/basic-dsa/array/counting-sort.md]: replace `id: @kjl98` with `id: counting-sort`

File: docs/basic-dsa/array/counting-sort.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+title: Added counting sort.md file
+sidebar_label: #333
```
Comment (ajay-dhangar) [inline docs/basic-dsa/array/counting-sort.md]: replace `sidebar_label: #333` with `sidebar_label: counting sort`

File: docs/basic-dsa/array/counting-sort.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
+---
+id: @kjl98
+title: Added counting sort.md file
```
Comment (ajay-dhangar) [inline docs/basic-dsa/array/counting-sort.md]: replace `title: Added counting sort.md file` with `title: Counting Sort Algorithm`

File: docs/basic-dsa/array/counting-sort.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+sidebar_label: #333 
+sidebar_position: ___
```
Comment (ajay-dhangar) [inline docs/basic-dsa/array/counting-sort.md]: The position in the sidebar where you want to add this item. Please enter a number between 1 and 1000.

File: docs/basic-dsa/array/counting-sort.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+			}
+		}
+	}
+
+	public static void main(String[] args) {
+		int arr[] = { 3, 6, 2, 1, 8, 7, 4, 5, 3, 1 };
+		countingSort(arr);
+	}
+}
+
```
Comment (ajay-dhangar) [inline docs/basic-dsa/array/counting-sort.md]: close your code example with ```` ``` ````

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: docs/basic-dsa/array/counting-sort.md (hunk 7 above)
Comment (kjl98) [reply docs/basic-dsa/array/counting-sort.md]: Done 👍, and thank you so much sir for guiding me.

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**